feat: Docker development environment

ojeytonwilliams opened the pull request:
This should make it a little easier to get started as a new contributor, as well as cut out 'works on my machine' bugs.

@freeCodeCamp/devdocs let me know what you think.  Does this all work as you expect?

simon04 commented on .github/CONTRIBUTING.md:
```diff
@@ -12,7 +12,8 @@ Want to contribute? Great. Please review the following guidelines carefully and
 7. [Other contributions](#other-contributions)
 8. [Coding conventions](#coding-conventions)
-9. [Questions?](#questions)
+9. [Developing in Docker](#developing-in-docker)
```
The section is missing in the file (has not been committed)?

simon04 commented on .env:
```diff
@@ -0,0 +1,2 @@
+USER_ID=1000
+GROUP_ID=1000
```
Maybe include a comment `# for Dockerfile-dev`?

simon04 commented on docker-compose.yml:
```diff
@@ -0,0 +1,13 @@
+version: '3.7'
```
Since we already have two `Dockerfile`s for running devdocs in production, I would not expect `docker-compose.yml` to be meant for development only.

simon04 commented on Dockerfile-dev:
```diff
@@ -0,0 +1,42 @@
+# a temporary directory to hold the lock files while the gems are installed.
+WORKDIR /devdocs/
+
+COPY Gemfile Gemfile.lock Rakefile /devdocs/
```
Running `docker build - < Dockerfile-dev` failed:

```
Step 9/14 : COPY Gemfile Gemfile.lock Rakefile /devdocs/
COPY failed: stat /var/lib/docker/tmp/docker-builder202400793/Gemfile: no such file or directory
```

(Not exactly sure how it is meant to be used, though…)

ojeytonwilliams commented:
Thanks for the review, Simon.  The latest commits should have addressed all your concerns - including adding the missing docs (thanks reflog :sweat:).

ojeytonwilliams commented:
Hey @simon04, I'm still working on this.  Right now using `thor: upload` doesn't work with AWS, so I need to figure it out.

tsengia commented:
@ojeytonwilliams and @simon04 Any updates on this? I can pick this up if you'd like. I cloned the repo and started making a devcontainer... and then realized you guys already started!
